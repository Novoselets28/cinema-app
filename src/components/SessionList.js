import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import FilmBox from './FilmBox';
import 'bootstrap/dist/css/bootstrap.min.css';
import { StyledContainer, StyledRow } from '../styled/SessionList';

const SessionList = () => {
  const films = useSelector((state) => state.sessionList.films);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch({type: 'FETCH_FILMS'});
  }, [dispatch]);

  return (
    <StyledContainer className="center-grid">
      <StyledRow>
      {films.length === 0 ? (
          <div>Loading...</div>
        ) : (
          films.map((filmReq) => (
            <FilmBox key={filmReq.id} {...filmReq} />
          ))
        )}
      </StyledRow>
    </StyledContainer>
  );
};


export default SessionList;
