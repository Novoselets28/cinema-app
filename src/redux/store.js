import { createStore, applyMiddleware, combineReducers } from 'redux';
import createSagaMiddleware from 'redux-saga';
import sessionTimeReducer from './reducers/reducersSessionTime';
import sessionListReducer from './reducers/reducersSessionList';
import sessionTimeSaga from './sagas/sagasSessionTime';
import sessionListSaga from './sagas/sagasSessionList';
import movieSessionsReducer from './reducers/reducersMovieSessions';
import movieSessionsSaga from './sagas/sagasMovieSessions';
import mainPageReducer from './reducers/reducersMainPage';
import mainPageSaga from './sagas/sagasMainPage';
import filmDescriptionReducer from './reducers/filmDescriptionReducer';
import filmDescriptionSaga from './sagas/filmDescriptionSaga';
import cinemaReducer from './reducers/reducersCinema';
import cinemaSaga from './sagas/cinemaSaga';

const sagaMiddleware = createSagaMiddleware();

const rootReducer = combineReducers({
  mainPage: mainPageReducer,
  sessionTime: sessionTimeReducer,
  sessionList: sessionListReducer,
  movieSessions: movieSessionsReducer,
  filmDescription: filmDescriptionReducer,
  cinema: cinemaReducer,

});

const store = createStore(
  rootReducer,
  applyMiddleware(sagaMiddleware)
);

sagaMiddleware.run(mainPageSaga);
sagaMiddleware.run(sessionTimeSaga);
sagaMiddleware.run(sessionListSaga);
sagaMiddleware.run(movieSessionsSaga);
sagaMiddleware.run(filmDescriptionSaga);
sagaMiddleware.run(cinemaSaga);

export default store;
