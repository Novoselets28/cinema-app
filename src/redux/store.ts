import { createStore, applyMiddleware, combineReducers } from 'redux';
import createSagaMiddleware from 'redux-saga';
import sessionTimeReducer, { sessionTimeSaga } from './ducks/sessionTime';
import filmDescriptionReducer, { filmDescriptionSaga } from './ducks/filmDescription';
import cinemaReducer, { cinemaSaga } from './ducks/cinema';
import sessionListReducer, { sessionListSaga } from './ducks/sessionList';
import movieSessionsReducer, { movieSessionsSaga } from './ducks/movieSessions';
import mainPageReducer, { mainPageSaga } from './ducks/mainPage';

export type RootState = ReturnType<typeof rootReducer>;

const sagaMiddleware = createSagaMiddleware();

export const rootReducer = combineReducers({
  mainPage: mainPageReducer,
  sessionTime: sessionTimeReducer,
  sessionList: sessionListReducer,
  movieSessions: movieSessionsReducer,
  filmDescription: filmDescriptionReducer,
  cinema: cinemaReducer,
});

const store = createStore(
  rootReducer,
  applyMiddleware(sagaMiddleware)
);

sagaMiddleware.run(mainPageSaga);
sagaMiddleware.run(sessionTimeSaga);
sagaMiddleware.run(sessionListSaga);
sagaMiddleware.run(movieSessionsSaga);
sagaMiddleware.run(filmDescriptionSaga);
sagaMiddleware.run(cinemaSaga);

export default store;