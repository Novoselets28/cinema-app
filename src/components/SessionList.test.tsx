import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store'; // Import your Redux store
import SessionList from './SessionList';
import { BrowserRouter } from 'react-router-dom';

describe('SessionList component', () => {
    it('renders loading message when films are empty', () => {
    const {container} = render(
        <Provider store={store}>
            <BrowserRouter>
                <SessionList date={''} />
            </BrowserRouter>
        </Provider>
    );

    const loadingMessage = screen.getByText('Loading...');
    expect(loadingMessage).toBeInTheDocument();
    expect(container).toMatchSnapshot()
    });

    it('renders FilmBox components when films are available', () => {
    const films = [
        { id: 1, Title: 'Film 1', Poster: 'Poster 1' },
        { id: 2, Title: 'Film 2', Poster: 'Poster 2' },
        { id: 3, Title: 'Film 3', Poster: 'Poster 3' },
    ];
    store.dispatch({ type: 'SET_FILMS_LIST', payload: films });

    const {container} = render(
        <Provider store={store}>
            <BrowserRouter>
                <SessionList date={''} />
            </BrowserRouter>
        </Provider>
    );
    films.forEach((film) => {
        const filmBoxTitle = screen.getByText(film.Title);

        expect(filmBoxTitle).toBeInTheDocument();
    });
    expect(container).toMatchSnapshot()
    });
})

