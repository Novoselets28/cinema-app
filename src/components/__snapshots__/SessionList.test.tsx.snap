// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionList component renders FilmBox components when films are available 1`] = `
<div>
  <div
    class="sc-fqkvVR jWBLYB center-grid container"
  >
    <div
      class="sc-dcJsrY iwazud row"
    >
      <div
        class="col-xxl-3 col-lg-4 col-md-6 col-12"
      >
        <div
          class="sc-eqUAAy bOpioT card"
        >
          <img
            alt="Film 1"
            class="card-img-top poster"
            src="Poster 1"
          />
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
            >
              Film 1
            </div>
            <button
              class="btn btn-primary"
              type="button"
            >
              View More
            </button>
          </div>
        </div>
      </div>
      <div
        class="col-xxl-3 col-lg-4 col-md-6 col-12"
      >
        <div
          class="sc-eqUAAy bOpioT card"
        >
          <img
            alt="Film 2"
            class="card-img-top poster"
            src="Poster 2"
          />
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
            >
              Film 2
            </div>
            <button
              class="btn btn-primary"
              type="button"
            >
              View More
            </button>
          </div>
        </div>
      </div>
      <div
        class="col-xxl-3 col-lg-4 col-md-6 col-12"
      >
        <div
          class="sc-eqUAAy bOpioT card"
        >
          <img
            alt="Film 3"
            class="card-img-top poster"
            src="Poster 3"
          />
          <div
            class="card-body"
          >
            <div
              class="card-title h5"
            >
              Film 3
            </div>
            <button
              class="btn btn-primary"
              type="button"
            >
              View More
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SessionList component renders loading message when films are empty 1`] = `
<div>
  <div
    class="sc-fqkvVR jWBLYB center-grid container"
  >
    <div
      class="sc-dcJsrY iwazud row"
    >
      <div>
        Loading...
      </div>
    </div>
  </div>
</div>
`;
